src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -304,1237 +234,653 @@ export default class Fms {
-     *
-     * This can be:
-     * - a directRouteString,
-     * - a procedureRouteString,
-     * - or combination of both directRouteStrings and procedureRouteString
-     *
-     * @for fms
-     * @method isValidRoute
-     * @param routeString {string}
-     * @param runway {string}
-     * @return {boolean}
-     */
-    isValidRoute(routeString, runway = '') {
-        const routeSegments = routeStringFormatHelper(routeString);
-
-        if (!routeSegments) {
-            return false;
-        }
-
-        for (let i = 0; i < routeSegments.length; i++) {
-            let isValid = false;
-            const segment = routeSegments[i];
-
-            if (RouteModel.isProcedureRouteString(segment)) {
-                isValid = this.isValidProcedureRoute(segment, runway);
-            } else if (RouteModel.isHoldRouteString(segment)) {
-                const fixName = extractFixnameFromHoldSegment(segment);
-
-                isValid = this._navigationLibrary.hasFix(fixName);
-            } else if (RouteModel.isVectorRouteString(segment)) {
-                const heading = extractHeadingFromVectorSegment(segment);
-                const isValidNumber = !isNaN(heading);
-                const isThreeDigits = heading.length === 3;
-
-                isValid = isValidNumber && isThreeDigits;
-            } else {
-                isValid = this._navigationLibrary.hasFix(segment);
-            }
-
-            if (!isValid) {
-                return false;
-            }
-        }
-
-        return true;
-    }
-
-    /**
-     * Determinines if the passed `routeString` is a valid procedure route.
-     *
-     * This can be either a SID or a STAR.
-     * A valid `procedureRouteString` is expected to be in the shape of:
-     * `ENTRY.PROCEDURE_NAME.EXIT`
-     *
-     * @for Fms
-     * @method isValidProcedureRoute
-     * @param routeString {string}
-     * @param runway {string}
-     * @param flightPhase {string}
-     * @return {boolean}
-     */
-    isValidProcedureRoute(routeString, runway, flightPhase = '') {
-        let routeStringModel;
-
-        // RouteModel will throw when presented with an invalid procedureRouteString,
-        // we only want to capture that here and continue on our way.
         try {
-            routeStringModel = new RouteModel(routeString);
+            nextRouteModel = new RouteModel(this._navigationLibrary, routeString);
         } catch (error) {
             console.error(error);
 
             return false;
         }
 
-        // flightPhase is unknown or unavailable so it must be extrapolated based on the `procedureId`.
-        if (flightPhase === '') {
-            flightPhase = this._translateProcedureNameToFlightPhase(routeStringModel.procedure);
-        }
-
-        // a `LegModel` already exists with this routeString
-        if (this.hasLegWithRouteString(routeStringModel.routeCode)) {
-            return true;
-        }
+        this._routeModel = nextRouteModel;
```

Comment:
is there a reason this is assigned here and with this intermediate var?  why not just:
```javascript
try {
  this._routeModel = new RouteModel(this._navigationLibrary, routeString);
} catch (error) {
  // ...
}
```